### DISCUSSION TITLE
Thoughts on making citations of sktime easier (and more up-to-date)?

### DISCUSSION THREAD

**Discussion by jmwhyte** [alan-turing-institute/sktime#2149] [General]
Researchers put a lot of time into code, and don't often get adequate acknowledgement for it.
I've seen this kind of discussion on [citing Python packages](https://discuss.python.org/t/convention-for-encouraging-citation-of-python-packages/1764/10).
I wondered if sktime had any interest in adopting something like a `__cite__` or (even better) `__bibtex__` attribute to make up-to-date information more readily available? This can only help citations.

**Comment by fkiraly** [comment]
Interesting idea!
The linked discussion does not seem to be a widely adopted standard though?
`sktime` does have a doi which can be cited: https://zenodo.org/record/6191159

What this also makes me wonder about is the curation of algorithms.
Some algorithms in `sktime` are reference implementations by the authors, so have their own primary citation attached. Conversely, optimally, reproducible research papers would cite version and identifier of an algorithm used in an experiment.
I wonder how either direction could be facilitated.

Sth like `cite(my_algorithm)` which produces a referencable signature? E.g., "algorithm [ID] from `sktime`, reference implementation (or re-implementation by) of algorithm X in [1], with default parameters/the following parameters"?

**Reply by jmwhyte** [reply]
@fkiraly I would certainly be happier about being able to cite the version of an algorithm I use. (We are collaborating "with our future selves" as well as others, so it's good to have details recorded.)

You might be right that the standard suggested hasn't been widely adopted, but there are cases, such as
`print(matplotlib.__bibtex__)`, which make the information readily available.

I asked the question because, whilst I found the page ["How to cite sktime"](https://www.sktime.org/en/stable/about/citation.html),
I noted that one reference is dated 2020, but you've recently released a new sktime version. So, I wondered if this information needs to be updated (possibly by a tedious manual process that sits outside of the package development at this stage)?

Perhaps I shall leave it to you and the other core developers to discuss this, and I'll take the advice from the webpage (cited above) until advised otherwise.
